mypy/subtypes.py

```diff
@@ -228,9 +228,14 @@ def visit_typeddict_type(self, left: TypedDictType) -> bool:
         elif isinstance(right, TypedDictType):
             if not left.names_are_wider_than(right):
                 return False
-            for (_, l, r) in left.zip(right):
+            for name, l, r in left.zip(right):
                 if not is_equivalent(l, r, self.check_type_parameter):
                     return False
+                # Non-required key is not compatible with a required key since indexing
+                # may fail.  Required key is not compatible with a non-required key
+                # since the prior doesn't support 'del' but the latter supports it.
```

Comment:
When I try to delete an item I always get `"A" has no attribute "__delitem__"`, e.g.
```py
A = TypedDict('A', {'x': int, 'y': str})
a: A = {'x': 0, 'y': ''}
del a['x']
```